### DISCUSSION TITLE
Should ELFIO use C++ 11 CMake var?

### DISCUSSION THREAD

**Discussion by serge1** [serge1/ELFIO#59] [Q&A] [answered]
The PR #57 suggests adding this var. Is it beneficial for the project?

**Comment by a4z** [comment, marked as answer]
I think this is absolutely fine, 

however, since there is nothing to compile, only tests and examples, and those should be disabled when you package the library, it has no effect for the cmake install, or, if you build a package.

For the _real_ compilation of elfio it will be used with what ever the cpp file of the user will be compiled with, and this will be very likely c++11, 14 , 17, or soon 20

So it might be a good idea to make that value configurable, so you can build and run the tests with any C++ standard you want cmake to configure with, and if non is given, fall back to 11.
Than it is easy to test that you do not use anything that was deprecated / disabled in a standard

**Reply by schultetwin1** [reply]
> however, since there is nothing to compile, only tests and examples, and those should be disabled when you package the library, it has no effect for the cmake install, or, if you build a package.

I agree. CMake should not set the C++ version when using ELFIO is as a library. We'd want to just inherit which ever value the user is already using.

My vote would be to only set the C++ standard in the examples and (maybe) test folders and not set it in the root folder
